### ISSUE TITLE
Upload multiple images for product in one go

### ISSUE THREAD

**Issue by ppumkin** [nopSolutions/nopCommerce#5123]
nopCommerce version: 
last 10 years worth apparently? 

Steps to reproduce the problem: 
You have to upload one image at a time. This is nonsense.

And such an easy fix.

**Comment by ppumkin** [comment]
All I found when searching around are articles from 10, 7, 5, even 2 years ago on how to do this???
I was a bit stumped - also nothing on the extension shop??

So this is quick and dirty fix. I don't care about alts and orders I just want to drag and drop multiples (or click upload and select multiples) I dont want to waste my life uploading photos, and im not going to use excel, or FTP or any thing like that.

If somebody gives me some guidance on how to properly do this,, as a plugin? PR.. something I will do it.

Otherwise for people who know how to edit the code.. I just did this

**Shared/EditorTemplates/Picture.cshtml**

```
<script>
    $(document).ready(function () {
        $("#@(clientId)").fineUploader({
            request: {
                endpoint: '@(Url.Content("~/Admin/Picture/AsyncUpload"))'
            },
            template: "@(clientId)-qq-template",
            multiple: true
        }).on("complete", function (event, id, name, responseJSON, xhr) {

            if (responseJSON.success) {

                var parts = window.location.href.split('/');
                var prdctId = parts.pop();

                 var postData = {
                                pictureId: responseJSON.pictureId,
                                displayOrder: 0,
                                overrideAltAttribute: '',
                                overrideTitleAttribute: '',
                     productId: prdctId
            };
                addAntiForgeryToken(postData);

                console.log(postData);
                            $.ajax({
                                cache: false,
                                type: "POST",
                                url: "@(Url.Action("ProductPictureAdd", "Product"))",
                                data: postData,
                                success: function (data, textStatus, jqXHR) {
                                    if (!data.Result) {
                                        return;
                                    }
                                    updateTable('#productpictures-grid');
                                },
                                error: function (jqXHR, textStatus, errorThrown) {
                                    //$("#productPictureAddAlert").click();
                                },
                                complete: function (jqXHR, textStatus) {
                                    $('#addProductPicture').attr('disabled', false);
                                }
                            });
            }
        });
    });
</script>
```

- This creates a "thumbnail" and picture ID- OK cool that code stays
- Change `mulitple : true` 
- Then on complete I used code from file below which then is just doing what that button used to 
- Every file then runs on its own "editor template" so to speak and it just works.

**Areas/Admin/Views/Product/_CreateOrUpdate.Pictures.cshtml**

```
@model ProductModel

<div class="panel-body">
    @if (Model.Id <= 0)
    {
<div class="panel panel-default">
    <div class="panel-body">
        @T("Admin.Catalog.Products.Pictures.SaveBeforeEdit")
    </div>
</div>}



    <div class="panel panel-default">
        <div class="panel-heading">
            @T("Admin.Catalog.Products.Pictures.AddNew")
        </div>
        <div class="panel-body">
            
            <div class="form-group">
                <div class="col-md-3">
                    <nop-label asp-for="AddPictureModel.PictureId" />
                </div>
                <div class="col-md-9">
                    <nop-editor asp-for="AddPictureModel.PictureId" />
                    <span asp-validation-for="AddPictureModel.PictureId"></span>
                </div>
            </div>
           
        </div>
    </div>

    <div class="sub-panel">

        @await Html.PartialAsync("Table", new DataTablesModel
        {
            Name = "productpictures-grid",
            UrlRead = new DataUrl("ProductPictureList", "Product", new RouteValueDictionary { [nameof(Model.ProductPictureSearchModel.ProductId)] = Model.ProductPictureSearchModel.ProductId }),
            UrlDelete = new DataUrl("ProductPictureDelete", "Product", null),
            UrlUpdate = new DataUrl("ProductPictureUpdate", "Product", null),
            Length = Model.ProductPictureSearchModel.PageSize,
            LengthMenu = Model.ProductPictureSearchModel.AvailablePageSizes,
            ColumnCollection = new List<ColumnProperty>
{
                    new ColumnProperty(nameof(ProductPictureModel.PictureUrl))
                    {
                        Title = T("Admin.Catalog.Products.Pictures.Fields.Picture").Text,
                        Render = new RenderCustom("renderPicturesColumnProductPictureUrl")
                    },
                    new ColumnProperty(nameof(ProductPictureModel.DisplayOrder))
                    {
                        Title = T("Admin.Catalog.Products.Pictures.Fields.DisplayOrder").Text,
                        Width = "150",
                        ClassName = NopColumnClassDefaults.CenterAll,
                        Editable = true,
                        EditType = EditType.Number
                    },
                    new ColumnProperty(nameof(ProductPictureModel.OverrideAltAttribute))
                    {
                        Title = T("Admin.Catalog.Products.Pictures.Fields.OverrideAltAttribute").Text,
                        Width = "200",
                        Editable = true,
                        EditType = EditType.String
                    },
                    new ColumnProperty(nameof(ProductPictureModel.OverrideTitleAttribute))
                    {
                        Title = T("Admin.Catalog.Products.Pictures.Fields.OverrideTitleAttribute").Text,
                        Width = "200",
                        Editable = true,
                        EditType = EditType.String
                    },
                    new ColumnProperty(nameof(ProductPictureModel.Id))
                    {
                        Title = T("Admin.Common.Edit").Text,
                        Width = "200",
                        ClassName =  NopColumnClassDefaults.Button,
                        Render = new RenderButtonsInlineEdit()
                    },
                    new ColumnProperty(nameof(ProductPictureModel.Id))
                    {
                        Title = T("Admin.Common.Delete").Text,
                        Width = "100",
                        Render = new RenderButtonRemove(T("Admin.Common.Delete").Text),
                        ClassName = NopColumnClassDefaults.Button
                    }
                }
        })

        <script>
            function renderPicturesColumnProductPictureUrl(data, type, row, meta) {
                return '<a href="' + row.PictureUrl + '" target="_blank"><img alt="' + row.PictureId + '" src="' + row.PictureUrl + '" width="150" /></a>';
            }
        </script>
    </div>

</div>
<nop-alert asp-alert-id="addProductPictureAlert" asp-alert-message="@T("Admin.Catalog.Products.Pictures.Alert.AddNew")" />
<nop-alert asp-alert-id="productPictureAddAlert" asp-alert-message="@T("Admin.Catalog.Products.Pictures.Alert.PictureAdd")" />
```


- I basically removed all the weird long winded stuff from this page
- Moved the upload / drop panel to the top of this section


![8xUh6L7LOv](https://user-images.githubusercontent.com/1249683/97504463-d5438100-196e-11eb-9979-894ea2914c16.gif)

**Comment by glhays** [comment]
Is everything functioning correctly on the public site?

**Comment by tombeuckelaere** [comment]
Hello all,

I think being able to upload multiple images at once is a good feature request. Multiple files/multiple images is something that is a standard in modern applications today, so NopCommerce should have it as well. It think some discussion can be held here to define the correct requirements of this functionality and then have it implemented properly?

**Workaround for people who don't want to alter code** 
There is currently a good extension available which allows import of multiple images at once for one or more products at once: https://www.nopcommerce.com/en/import-manager-20-foxnetsoftcom. I'm using it myself very often.

**Comment by ppumkin** [comment]
Cool - Well I am happy to make a free contribution into the core project only as to benefit everybody.. because frankly its a bit strange its like the way it is.. for so many years?

I just need the project coordinator to say what and how they expect this to work.

My proposal.

- Convert the existing partial to accept multiple files 
- I would normally say to not convert things but given the circumstance this existing functionality is well over due by many years
- Remove the "thumbnail" image and make a nicer looking drag and drop area
- As in my animation process all the images and add them into the list

**Comment by ppumkin** [comment]
> Is everything functioning correctly on the public site? I agree that would be a great improvement.

Yes 😄

**Comment by Tjax33** [comment]
Is this ready to download? 
This is a very much needed feature for my company.

**Comment by DmitriyKulagin** [comment]
Closed #5123
